cudapoa/tests/Test_CudapoaNW.cu

```diff
@@ -185,86 +186,81 @@ std::vector<NWTestPair> getNWTestCases()
-    memset((void**)outgoing_edge_count, 0, batch_size.max_nodes_per_graph * sizeof(uint16_t));
-    memset((void**)node_id_to_pos, 0, batch_size.max_nodes_per_graph * sizeof(SizeT));
-    memset((void**)scores, 0, batch_size.max_nodes_per_graph * batch_size.matrix_sequence_dimension * sizeof(int16_t));
+    memset(incoming_edge_count, 0, batch_size.max_nodes_per_graph * sizeof(uint16_t));
+    memset(outgoing_edge_count, 0, batch_size.max_nodes_per_graph * sizeof(uint16_t));
+    memset(node_id_to_pos, 0, batch_size.max_nodes_per_graph * sizeof(int16_t));
+    memset(scores, 0, batch_size.max_nodes_per_graph * batch_size.matrix_sequence_dimension * sizeof(int16_t));
 
     //calculate edge counts on host
     obj.get_graph_buffers(incoming_edges, incoming_edge_count,
                           outgoing_edges, outgoing_edge_count,
                           nodes, &graph_count,
                           graph, node_id_to_pos);
     obj.get_read_buffers(read, &read_count);
-    gap_score      = BasicNW::gap_score_;
-    mismatch_score = BasicNW::mismatch_score_;
-    match_score    = BasicNW::match_score_;
+    int32_t gap_score      = BasicNW::gap_score_;
+    int32_t mismatch_score = BasicNW::mismatch_score_;
+    int32_t match_score    = BasicNW::match_score_;
 
     //call the host wrapper of nw kernel
-    runNW<SizeT>(nodes,
-                 graph,
-                 node_id_to_pos,
-                 graph_count,
-                 incoming_edge_count,
-                 incoming_edges,
-                 outgoing_edge_count,
-                 read,
-                 read_count,
-                 scores,
-                 BatchConfig().matrix_sequence_dimension,
-                 alignment_graph,
-                 alignment_read,
-                 gap_score,
-                 mismatch_score,
-                 match_score,
-                 aligned_nodes);
+    runNW(nodes,
+          graph,
+          node_id_to_pos,
+          graph_count,
+          incoming_edge_count,
+          incoming_edges,
+          outgoing_edge_count,
+          read,
+          read_count,
+          scores,
+          batch_size.matrix_sequence_dimension,
+          alignment_graph,
+          alignment_read,
+          gap_score,
+          mismatch_score,
+          match_score,
+          aligned_nodes);
 
     GW_CU_CHECK_ERR(cudaDeviceSynchronize());
 
     //input and output buffers are the same ones in unified memory, so the results are updated in place
     //results are stored in alignment_graph and alignment_read; return string representation of those
-    auto res = std::make_pair(genomeworks::stringutils::array_to_string<SizeT>(alignment_graph, *aligned_nodes, ","),
-                              genomeworks::stringutils::array_to_string<SizeT>(alignment_read, *aligned_nodes, ","));
+    auto res = std::make_pair(genomeworks::stringutils::array_to_string<int16_t>(alignment_graph, *aligned_nodes, ","),
```

Comment:
Optional: template argument `<int16_t>` not needed - let the compiler deduce it automatically from the `alignment_graph` argument.

Reply:
right, that's result of using replace all! removed.